### PR TITLE
Change the backoff based on the ratelimit header provided by contentful 

### PR DESCRIPTION
**Pull request by Khaledgarbaya**

### CODE REVIEW

**Comment by Khaledgarbaya** [comment]
@stefanjudis whenever you have time :)

**Comment by stefanjudis** [comment]
@Khaledgarbaya What's this random factor about?

**Comment by Khaledgarbaya** [comment]
@stefanjudis it is a random buffer, suggested by the backend engineers to avoid spamming the api

File: lib/create-backoff.js [hunk 1]
```diff
@@ -12,7 +12,10 @@ export default function createBackoff (maxRetries) {
   let attempt = 0
   return function maybeRetry (error, retry) {
     if (attempt < maxRetries) {
-      return promisedWait(Math.pow(2, attempt++) * 1000).then(retry)
+      let waitTime = (error.headers && error.headers['X-Contentful-RateLimit-Reset']) || Math.pow(2, attempt)
+      attempt++
+      waitTime *= 1000 * ((Math.random() * 1.2) + 1) // add a randomized buffer before recalling the api
```
Comment (stefanjudis) [inline lib/create-backoff.js]: 👍 for the comment but I think it should explain more why it's there.
I understand the code snippet quickly but the why is more important for me (our slack conversation).

**Comment by stefanjudis** [comment]
🚢👍

**Pull request merged**